src/app/modules/shared/components/details-fields/details-fields.component.ts

```diff
@@ -197,29 +197,56 @@ export class DetailsFieldsComponent implements OnChanges, OnInit {
     this.closeModal?.nativeElement?.click();
   }
 
+  startDateToSubmit(){
+    const startDate = this.entryForm.value.start_date;
+    const initialStartDate = this.entryToEdit.start_date;
+    const updatedStartDate = new Date(`${startDate}T${this.entryForm.value.start_hour.trim()}`).toISOString();
+    const initialStartHour = formatDate(get(this.entryToEdit, 'start_date', '00:00'), 'HH:mm', 'en');
+    const updatedStartHour = this.entryForm.value.start_hour;
+    const startHourHasNotChanged = updatedStartHour === initialStartHour;
+    const result = startHourHasNotChanged ? initialStartDate : updatedStartDate;
+    return result;
+  }
+
+  endDateToSubmit(){
+    const endDate = this.entryForm.value.end_date;
+    const initialEndDate = this.entryToEdit.end_date;
+    const updatedEndDate = new Date(`${endDate}T${this.entryForm.value.end_hour.trim()}`).toISOString();
+    const initialEndHour = formatDate(get(this.entryToEdit, 'end_date', '00:00'), 'HH:mm', 'en');
+    const updatedEndHour = this.entryForm.value.end_hour;
+    const endDateHasNotChanged = updatedEndHour === initialEndHour;
+    const result = endDateHasNotChanged ? initialEndDate : updatedEndDate;
+    return result;
+  }
+
   onSubmit() {
     if (this.entryForm.invalid) {
       this.toastrService.warning('Make sure to select a project and activity');
       return;
     }
-    const startDate = this.entryForm.value.start_date;
-    const endDate = this.entryForm.value.end_date;
+
+    const startDateToSubmit = this.startDateToSubmit();
+    const endDateToSubmit = this.endDateToSubmit();
+
     const entry = {
       project_id: this.entryForm.value.project_id,
       activity_id: this.entryForm.value.activity_id,
       technologies: get(this, 'selectedTechnologies', []),
       description: this.entryForm.value.description,
-      start_date: new Date(`${startDate}T${this.entryForm.value.start_hour.trim()}`).toISOString(),
-      end_date: new Date(`${endDate}T${this.entryForm.value.end_hour.trim()}`).toISOString(),
+      start_date: startDateToSubmit,
+      end_date: endDateToSubmit,
       uri: this.entryForm.value.uri,
       timezone_offset: new Date().getTimezoneOffset(),
     };
     if (this.goingToWorkOnThis) {
       delete entry.end_date;
     }
-    const isStartDateInTheFuture = moment(startDate).isAfter(moment());
-    const isEndDateInTheFuture = moment(endDate).isAfter(moment());
-    if (isStartDateInTheFuture || isEndDateInTheFuture) {
+
+    const isStartDateInTheFuture = moment(startDateToSubmit).isAfter(moment());
+    const isEndDateInTheFuture = moment(endDateToSubmit).isAfter(moment());
+    const timeEntryIsInTheFuture = isStartDateInTheFuture || isEndDateInTheFuture;
+
+    if (timeEntryIsInTheFuture) {
```

Comment:
I remember we made a change to extract `timeEntryIsInTheFuture` into a function. I guess that caused issues in existing tests and you ended up doing it like this. Am I right, or  is it an error?

Reply:
I just tried the function of not allowing a "time entry" in the future and it allowed me to add. It ignores the way we put it so I left it as it was. but with the tests, there was no problem.

Reply:
It's ok. Then, keep in mind that, when we may want to retake that little refactor.